gk/main.c

```diff
@@ -294,6 +310,7 @@ gk_process_request(struct flow_entry *fe, struct ipacket *packet,
 	} else {
 		fe->u.request.last_priority = priority;
 		fe->u.request.allowance = START_ALLOWANCE - 1;
+		fe->expire_at = calc_request_expire_at(now);
```

Comment:
As we are introducing the expiration time for flows with ``REQUEST`` state, should we also deal with the flow entry expiration within ``gk_process_request()``? If so, we need to think about what actions should be taken for those flows.

Reply:
A request entry is the fallback state. That is, when an entry in another state expires, the entry becomes a request entry. Thus, there's no fallback for an expired request entry. Moreover, the code properly deals with expired request entries: (1) `gk_proc()` removes expired (request) entries and (2) `gk_process_request()` calculates a new priority. To better understand case (2), remember that an expired request entry will have `priority > fe->u.request.last_priority` in `gk_process_request()`.

In order words, I think we don't need to write more code to handle expired request entries.